Cython/Compiler/TypeSlots.py

```diff
@@ -424,7 +424,7 @@ def __init__(self, slot_name, **kargs):
         InternalMethodSlot.__init__(self, slot_name, **kargs)
 
     def slot_code(self, scope):
-        if not scope.needs_gc():
+        if not scope.needs_gc() or scope.parent_type.is_external:
             return "0"
```

Comment:
This seems the opposite of the described intention – setting the slot to `0` if it's a potentially incomplete type.

Reply:
These functions use `"0"` to indicate that it's unknown. So in this case it means that we don't try to make a direct call to the external type's gc function. I've updated the comment but I think this is right.